File: q1704/q1704.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        vowels = set(['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'])
        count = 0

        # Increment count in the first half
        for i in range(0, len(s) // 2):
            if s[i] in vowels: count += 1

        # Decrement count in the second half
        for i in range(len(s) // 2, len(s)):
            if s[i] in vowels: count -= 1
        
        return count == 0

File: q1704/q1704.java
class Solution {
    public boolean halvesAreAlike(String s) {
        Set<Character> vowels = new HashSet(Arrays.asList('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'));
        int count = 0;
        int n = s.length();

        // Increment count in the first half
        for (int i = 0; i < n / 2; i++) {
            if (vowels.contains(s.charAt(i)))
                count++;
        }

        // Decrement count in the second half
        for (int i = n / 2; i < n; i++) {
            if (vowels.contains(s.charAt(i)))
                count--;
        }

        return count == 0;
    }
}